util/unifyfs/src/unifyfs-rm.c

```diff
@@ -483,58 +485,132 @@ static int slurm_read_resource(unifyfs_resource_t* resource)
 static size_t construct_server_argv(unifyfs_args_t* args,
                                     char** server_argv)
 {
-    size_t argc;
+    size_t argc = 0;
     char number[16];
 
-    if (server_argv != NULL) {
-        if (args->server_path != NULL) {
-            server_argv[0] = strdup(args->server_path);
-        } else {
-            server_argv[0] = strdup(BINDIR "/unifyfsd");
-        }
-    }
-    argc = 1;
-
-    if (args->debug) {
+    switch (args->arg_type) {
+    case UNIFYFS_CONSTRUCT_ARGS_UNIFYFSD:
         if (server_argv != NULL) {
-            server_argv[argc] = strdup("-v");
-            snprintf(number, sizeof(number), "%d", args->debug);
-            server_argv[argc + 1] = strdup(number);
+            if (args->server_path != NULL) {
+                server_argv[0] = strdup(args->server_path);
+            } else {
+                server_argv[0] = strdup(BINDIR "/unifyfsd");
+            }
         }
-        argc += 2;
-    }
+        argc = 1;
 
-    if (args->cleanup) {
-        if (server_argv != NULL) {
-            server_argv[argc] = strdup("-C");
-        }
-        argc++;
-    }
+	if (args->debug) {
+            if (server_argv != NULL) {
+	        server_argv[argc] = strdup("-v");
+		snprintf(number, sizeof(number), "%d", args->debug);
+		server_argv[argc + 1] = strdup(number);
+            }
+	    argc += 2;
+	}
 
-    if (args->consistency != UNIFYFS_CM_LAMINATED) {
+	if (args->cleanup) {
+            if (server_argv != NULL) {
+	        server_argv[argc] = strdup("-C");
+            }
+	    argc++;
+	}
+
+	if (args->consistency != UNIFYFS_CM_LAMINATED) {
+            if (server_argv != NULL) {
+	        server_argv[argc] = strdup("-c");
+		server_argv[argc + 1] =
+		    strdup(unifyfs_cm_enum_str(args->consistency));
+            }
+	    argc += 2;
```

Comment:
Check you indents.  There's a couple of places where you have mixed tabs/spaces.